lib/grant-types/refresh-token-grant-type.js

```diff
@@ -142,6 +145,33 @@ class RefreshTokenGrantType extends AbstractGrantType {
 
     return this.model.saveToken(token, client, user);
   }
+
+  getScope (request, token) {
```

Comment:
can you shortly elaborate on this, why it replaced the former behaviour?

Reply:
The former behaviour ignored the `scope` parameter passed with the data of the `refresh_token` grant type 

The scope was directly reused from the refresh token

`return this.saveToken(token.user, client, token.scope);`

now we use the scope which is parsed and validated from this function

```
const scope = this.getScope(request, token);
return this.saveToken(token.user, client, scope);
```

This function 
- takes the requested scope and the original scope
- parses them as described in the spec
- checks if there are no extra scopes passed which were not included in the previous token